File: apps/mobile/lib/services/support_service.dart
import '../core/network/api_client.dart';
import '../core/storage/secure_store.dart';

class SupportService {
  SupportService({SecureStore? store})
      : _client = ApiClient(store ?? SecureStore());

  final ApiClient _client;

  Future<void> sendReport({
    required String category,
    required String message,
    required bool includeLogs,
  }) async {
    await _client.post(
      '/support',
      auth: true,
      body: {
        'type': 'report',
        'category': category,
        'message': message,
        'includeLogs': includeLogs,
      },
    );
  }

  Future<void> sendFeedback({
    required double score,
    required String message,
    required bool allowContact,
  }) async {
    await _client.post(
      '/support',
      auth: true,
      body: {
        'type': 'feedback',
        'message': message,
        'score': score,
        'allowContact': allowContact,
      },
    );
  }

  Future<void> sendHelp({
    required String message,
  }) async {
    await _client.post(
      '/support',
      auth: true,
      body: {
        'type': 'help',
        'message': message,
      },
    );
  }
}

File: apps/web/src/services/support-service.ts
import { apiClient } from '../adapters/api-client';

class SupportService {
  async sendReport(input: {
    category: string;
    message: string;
    includeLogs: boolean;
  }) {
    await apiClient.post('/support', {
      auth: true,
      body: {
        type: 'report',
        category: input.category,
        message: input.message,
        includeLogs: input.includeLogs,
      },
    });
  }

  async sendFeedback(input: {
    score: number;
    message: string;
    allowContact: boolean;
  }) {
    await apiClient.post('/support', {
      auth: true,
      body: {
        type: 'feedback',
        score: input.score,
        message: input.message,
        allowContact: input.allowContact,
      },
    });
  }

  async sendHelp(input: { message: string }) {
    await apiClient.post('/support', {
      auth: true,
      body: {
        type: 'help',
        message: input.message,
      },
    });
  }
}

export const supportService = new SupportService();
